addon_service/common/waterbutler_compat.py

```diff
@@ -24,15 +24,28 @@ class JSONAPIMeta:
 
     def _credentials_for_waterbutler(self, configured_storage_addon):
         _creds_data = configured_storage_addon.credentials
+        imp = get_storage_addon_instance__blocking(
+            configured_storage_addon.imp_cls,
+            configured_storage_addon.base_account,
+            configured_storage_addon.config,
+        )
+        wb_config = async_to_sync(imp.build_wb_config)()
```

Comment:
I think this may cause performance issues, 
I'd implement some sort of caching for `_config_for_waterbutler` method